library/core/src/iter/traits/iterator.rs

```diff
@@ -1825,10 +1825,20 @@ pub trait Iterator {
         Inspect::new(self, f)
     }
 
-    /// Borrows an iterator, rather than consuming it.
+    /// Iterator adapter that turns ownership-taking methods into mutating methods.
```

Comment:
```suggestion
    /// Creates a "by reference" adapter for this instance of `Iterator`.
```

I know this is a little vague, but the way you've written it reads awkwardly. The original seems fine as well with the new explanation. It's hard to fit exactly what we want to say into one sentence, so I think being general is fine.

Reply:
Alright, I agree that my phrasing is perhaps not good, and the other explanation is now better, so the summary can be simpler. Going off your "by reference" iterator, how about something like: "Iterator adapter that passes on method calls to the original iterator by mutable reference."?

Reply:
How about we keep my suggestion for now to mirror `io::{Read, Write}::by_ref` and `Allocator::by_ref`, and you can open a separate PR to discuss changing it? I don't want to get too hung up on this.

Reply:
Alright, done.